src/Compilers/CSharp/Test/Semantic/Semantics/NullableContextTests.cs

```diff
@@ -413,8 +429,777 @@ void verify(CSharpParseOptions parseOptions, bool expectedFlowState, params stri
+                //     partial void F2(object y = null) { }
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(5, 32),
+                // (7,28): warning CS1066: The default value specified for parameter 'x' will have no effect because it applies to a member that is used in contexts that do not allow optional arguments
+                //     partial void F1(object x = null) { }
+                Diagnostic(ErrorCode.WRN_DefaultValueForUnconsumedLocation, "x").WithArguments("x").WithLocation(7, 28));
+
+            var actualAnalyzedKeys = GetNullableDataKeysAsStrings(comp.NullableAnalysisData, requiredAnalysis: true);
+            AssertEx.Equal(new[] { ".cctor", ".ctor", "= null", "= null", "F2" }, actualAnalyzedKeys);
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_MethodBodySemanticModel_01()
+        {
+            var source =
+@"class Program
+{
+#nullable enable
+    static object F1(object o1)
+    {
+        if (o1 == null) { }
+        return o1;
+    }
+#nullable disable
+    static object F2(object o2)
+    {
+        if (o2 == null) { }
+        return o2;
+    }
+}";
+
+            var comp = CreateCompilation(source);
+            comp.NullableAnalysisData = new ConcurrentDictionary<object, NullableWalker.Data>();
+            var syntaxTree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(syntaxTree);
+            var returnStatements = syntaxTree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().ToArray();
+
+            var syntax = returnStatements[0];
+            Assert.Equal("return o1;", syntax.ToString());
+            var typeInfo = model.GetTypeInfo(syntax.Expression);
+            Assert.Equal(Microsoft.CodeAnalysis.NullableFlowState.MaybeNull, typeInfo.Nullability.FlowState);
+
+            syntax = returnStatements[1];
+            Assert.Equal("return o2;", syntax.ToString());
+            typeInfo = model.GetTypeInfo(syntax.Expression);
+            Assert.Equal(Microsoft.CodeAnalysis.NullableFlowState.None, typeInfo.Nullability.FlowState);
+
+            var actualAnalyzedKeys = GetNullableDataKeysAsStrings(comp.NullableAnalysisData, requiredAnalysis: true);
+            AssertEx.Equal(new[] { "F1" }, actualAnalyzedKeys);
+        }
+
+        [Fact]
+        [WorkItem(49746, "https://github.com/dotnet/roslyn/issues/49746")]
+        public void AnalyzeMethodsInEnabledContextOnly_MethodBodySemanticModel_02()
+        {
+            var source =
+@"class A
+{
+#nullable enable
+    object F;
+    A(object o1)
+#nullable disable
+    {
+        if (o1 == null) { }
+        F = o1;
+    }
+}
+class B
+{
+#nullable enable
+    object F;
+#nullable disable
+    B(object o2)
+    {
+        if (o2 == null) { }
+        F = o2;
+    }
+}";
+
+            var comp = CreateCompilation(source);
```

Comment:
Should we verify diagnostics in this test and other new tests?

Reply:
> Should we verify diagnostics in this test and other new tests?

I'd rather not, so it's clear the entries in `comp.NullableAnalysisData` are from the `SemanticModel` only.

---
In reply to: [544706421](https://github.com/dotnet/roslyn/pull/49953#discussion_r544706421) [](ancestors = 544706421)